### PR TITLE
Default to a ping lifetime during conversion

### PR DESCRIPTION
**Pull request by badboy**
### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: glean-core/src/common_metric_data.rs [hunk 1]
```diff
@@ -30,15 +28,13 @@ impl Lifetime {
...
+    fn from(value: i32) -> Lifetime {
         match value {
-            0 => Ok(Lifetime::Ping),
-            1 => Ok(Lifetime::Application),
-            2 => Ok(Lifetime::User),
-            _ => Err(()),
+            0 => Lifetime::Ping,
+            1 => Lifetime::Application,
+            2 => Lifetime::User,
+            _ => Lifetime::Ping,
```
Comment (Dexterp37) [inline glean-core/src/common_metric_data.rs]: Mh, what's the rationale behind this? In general, we try not to have "fallback" behaviours in Glean, but rather have clear erroring paths. What are we adjusting here?

Comment (badboy) [reply glean-core/src/common_metric_data.rs]: Hm, I see. 
I was trying to simplify the FFI component.
So far we did error report and still fallback to `Lifetime::Ping`.

I can see the use of "log error and bail out" though. Let me try something.

Comment (Dexterp37) [reply glean-core/src/common_metric_data.rs]: Cheers! Thanks for 🐻 ing with me :P

**Comment by badboy** [comment]
@Dexterp37 What do you think about this?

**Pull request merged**